docs/src/main/sphinx/connector/sqlserver.rst

```diff
@@ -49,20 +49,22 @@ properties files.
 Connection security
 ^^^^^^^^^^^^^^^^^^^
 
-If you have TLS configured with a globally-trusted certificate installed on your
-data source, you can enable TLS between your cluster and the data
-source by appending a parameter to the JDBC connection string set in the
-``connection-url`` catalog configuration property.
+The JDBC driver, and therefore the connector, automatically use Transport Layer
+Security (TLS) encryption and certificate validation. This requires a suitable
+TLS certificate configured on your Trino cluster nodes and your SQL Server
+database host.
```

Comment:
```suggestion
Security (TLS) encryption and certificate validation. This requires a working
TLS configuration/setup on your SQL Server database host.
```

trino cluster cares about TLS only if someone uses self-signed certs but at that point I'd assume they know what they are doing and they'll also need some more config options for the JDBC driver and SQL Server to accept self-signed certificates.

Let's make the 99% of setups easier and for the remaining 1% we can either direct them to SQL Server docs or something that explains what's needed to get self-signed certs working with Java in general and Trino specificallly. It's a separate work-stream though.

Reply:
The problem is that most corporation use custom certs from their own CA.. so the 99% is more like 50% .. 

Basically the cert in the JVM on the coordinator and the workers needs to be trusted by the cert of the database host. 

We definitely need to say that and I what we have now is my attempt to keep it simple but not omit this crucial info. Theoretically if both certs are globally trusted it should just work .. but I have not tested that and when it comes to TLS and certs it seems like its never that easy..

Reply:
I believe you are thinking of mTLS where both server and client present their own certs and verify each other.

But sql server tls is about having a TLS cert on SQL Server and getting the JVM to trust it.

Agreed about self-signed certs though.

Reply:
So you are saying in this case. .. only Trino has to trust the cert provided by the database .. but not the other way around?

Reply:
Yes. That is how TLS works in general. In case of intra-cluster security we need both Trino nodes to trust each other + client to trust Trino server since there are "two" pairs of comms to secure. node <-> node and client <-> co-ordinator.

But with data sources TLS applies only to data-source -> node and it's unidirectional. You can configure mTLS (where both sides verify each other) with Kafka for example but I won't worry about documenting that unless someone asks.

Reply:
interesting.. thanks for explaining that @hashhar  .. I updated the wording accordingly.